spyder/plugins/editor/panels/codefolding.py

```diff
@@ -248,6 +250,28 @@ def paintEvent(self, event):
         # on the folding panel.
         super(FoldingPanel, self).paintEvent(event)
         painter = QPainter(self)
+        if not self._display_folding and not self._key_pressed:
+            if any(self.folding_status.values()):
+                for info in self.editor.visible_blocks:
+                    top_position, line_number, block = info
+                    if line_number in self.folding_regions:
+                        collapsed = self.folding_status[line_number]
+                        line_end = self.folding_regions[line_number]
+                        mouse_over = self._mouse_over_line == line_number
+                        if collapsed:
+                            self._draw_fold_indicator(
+                                top_position, mouse_over, collapsed, painter)
+                            # check if the block already has a decoration,
+                            # it might have been folded by the parent
+                            # editor/document in the case of cloned editor
+                            for deco in self._block_decos:
+                                if deco.block == block:
+                                    # no need to add a deco, just go to the
+                                    # next block
+                                    break
+                            else:
+                                self._add_fold_decoration(block, line_end)
+            return
```

Comment:
The code you added in these lines is very, very similar to the code to draw fold triggers from like 10 lines below. Do you see a way to refactor both and have a single method here and there?